### PR TITLE
Plane config

### PR DESCRIPTION
**Pull request by Vladou73**
- creation of model PlaneConfiguration (class + properties and methods)
- plane constants added in OGREConstants.js
- API route to retrieve data in DB from table plane_configuration

### CODE REVIEW

File: server/app/OGREConstants.js [hunk 1]
```diff
@@ -4,7 +4,9 @@ const planeConstants = {
     DISTANCE_1_RIDE : 14200,
     PASSENGERS_BY_PLANE : 416,
     CALORIFIC_VALUE : 10,
-    CONSUMPTION_BY_PASSENGER : (this.CONSUMPTION_1_RIDE * this.CALORIFIC_VALUE / this.PASSENGERS_BY_PLANE) * 2
+    get CONSUMPTION_BY_PASSENGER() {
```
Comment (remi-boivin) [inline server/app/OGREConstants.js:7]: You should remove new line after the function

File: server/app/OGREConstants.js [hunk 2]
```diff
@@ -19,7 +21,7 @@ const OGREConstants = {
     daysPerYear : 365
 }
 
-
+console.log(OGREConstants.planeConstants.CONSUMPTION_BY_PASSENGER * 2);
 // const CarConstants = {
 //     CONSUMPTION100KM: 7,
 //     CALORIFIC_VALUE: 10,
```
Comment (remi-boivin) [inline server/app/OGREConstants.js:26]: You must remove commented code. It's dirty to commit commented code

File: server/app/mocked_players.js [hunk 3]
```diff
@@ -36,4 +37,17 @@ const players = {
...
 }
+
+let planeData1 = {
+    id : 1,
+    distancePerYear : 5000
+}
+
+let planeConfig1 = new PlaneConfiguration(planeData1);
+planeConfig1.computeEnergyConsumptionPerYear();
+// planeConfig1.computeEnergyConsumptionPerDay();
```
Comment (remi-boivin) [inline server/app/mocked_players.js]: Same here you need to remove commented code

File: server/app/mocked_players.js [hunk 4]
```diff
@@ -36,4 +37,17 @@ const players = {
...
+
+console.log(planeConfig1);
+
```
Comment (remi-boivin) [inline server/app/mocked_players.js]: You may remove a new line

File: server/app/mocked_players.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
+const OGREConstant = require('./OGREConstants');
 const PersonaOgre = require('./models/PersonaOgre')
 const CarConfiguration = require('./models/CarConfiguration')
-const OGREConstant = require('./OGREConstants')
+const PlaneConfiguration = require('./models/PlaneConfiguration');
 
 // first mock
```
Comment (remi-boivin) [inline server/app/mocked_players.js:6]: I don't sure the comment is useful. Maybe you should either remove it or details it

File: server/app/controllers/playerController.js [hunk 6]
```diff
@@ -27,11 +27,11 @@ playerController.allPlayersBis = async (_, response, next) => {
...
 
-playerController.getPlaneConfig = async (planeId, response, next) => {
-    console.log('enter playerController.getPlaneConfig');
+playerController.getPlaneConfigs = async (_, response, next) => {
+    console.log('enter playerController.getPlaneConfigs');
     try {
-        const planeConfig = await PlaneConfiguration.findOne(planeId);
-        response.json(planeConfig);
+        const planeConfigs = await PlaneConfiguration.findAll();
+        response.json(planeConfigs);
```
Comment (remi-boivin) [inline server/app/controllers/playerController.js:34]: We should check if we found data and return either a success or a error message with corresponding status code

File: server/app/models/PlaneConfiguration.js [hunk 7]
```diff
@@ -22,6 +23,23 @@ class PlaneConfiguration {
...
         return this.energyConsumptionPerDay;
     }
+
+    static async findAll() {
+        const query = `SELECT * FROM plane_config;`;
+        try {
+            const result = await db.query(query);
+            let plane_configs = [];
+            for (const row of result.rows) {
+                // on transforme la donnée brute de la query sql en instance de PlaneConfiguration
```
Comment (remi-boivin) [inline server/app/models/PlaneConfiguration.js]: Please translate the comment into English for a better understanding

File: server/app/OGREConstants.js [hunk 8]
```diff
@@ -4,7 +4,9 @@ const planeConstants = {
     DISTANCE_1_RIDE : 14200,
     PASSENGERS_BY_PLANE : 416,
     CALORIFIC_VALUE : 10,
-    CONSUMPTION_BY_PASSENGER : (this.CONSUMPTION_1_RIDE * this.CALORIFIC_VALUE / this.PASSENGERS_BY_PLANE) * 2
+    get CONSUMPTION_BY_PASSENGER() {
+        return (this.CONSUMPTION_1_RIDE * this.CALORIFIC_VALUE / this.PASSENGERS_BY_PLANE) * 2
+    }
```
Comment (remi-boivin) [inline server/app/OGREConstants.js:9]: Good work

File: server/app/models/PlaneConfiguration.js [hunk 9]
```diff
@@ -1,4 +1,5 @@
 const OGREConstants = require("../OGREConstants");
+const db = require('../database');
```
Comment (remi-boivin) [inline server/app/models/PlaneConfiguration.js:2]: You need to change the path. database is in config folder

**Review by remi-boivin — CHANGES_REQUESTED** [review]

**Comment by Vladou73** [comment]
this pull request is outdated. It has been replaced by the pull request #8

**Pull request closed without merging**